mrp_lot_production_date/models/mrp_production.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def action_generate_serial(self):
+        res = super().action_generate_serial()
+        if self.lot_producing_id.product_id.use_production_date:
+            self.lot_producing_id.production_date = fields.Datetime.now()
```

Comment:
This will set the production to when the user clicks on the button in the manufacturing order. IMO that's good as default if the field is required (cf my comment https://github.com/OCA/stock-logistics-workflow/pull/1162#pullrequestreview-1237598249 ).
But depending on each company's process the lot number could be set before the the manufacturing is processed, in which case the production date probably needs to be overwritten later in their process.

Reply:
In our use-case we let the MO creating the lot automatically (`action_generate_serial` is called by the production wizard).
I could move this in the wizard process method instead, could be a better default behavior...
thanks!

Reply:
NOTE to myself: indicate clearly in the readme that the date will be set at the validation of the MO (and not at lot creation).